Automattic/woocommerce-payments
# refactor: ECE refreshes blocks UI in all scenarios

Fixes #

#### Changes proposed in this Pull Request









I'd like to propose a slight alternative to https://github.com/Automattic/woocommerce-payments/pull/9888 .

I noticed that if you have multiple shipping options, and you interact with them, when the dialog is dismissed the selected shipping option might not be accurate.
Let's take this example:
![2024-12-20 17 43 29](https://github.com/user-attachments/assets/60eb7c75-5d6c-4ee6-a251-5b3681e6316a)

The solution I'm proposing creates a "bogus" endpoint in the backend, which can be hit w/ the Block API's `extensionCartUpdate`, so that the UI is refreshed with new data (which includes the selected shipping rate, and the anonymized address for UK and CA.

This only applies to the blocks checkout, because it gives us this opportunity.

#### Testing instructions





- Ensure you have GooglePay/ApplePay enabled in your WooPayments settings
- Go to a block-based cart or checkout pages
- Click on the GooglePay/ApplePay button
- Select a different shipping rate
- Dismiss the dialog
- The selected shipping rate should be honored on the UI

> [!NOTE]
> In case of UK or CA addresses, the ZIP code displayed will be the same that would be displayed if the customer just refreshed the page. We were preventing the submission of UK and CA addresses only because the API would fail. But the address was updated anyways.

-------------------


**Post merge**
